beluga_system_tests/test/test_system_new.cpp

```diff
@@ -170,13 +173,8 @@ auto particle_filter_test(
         probability_estimator.reset();
       }
 
-      auto make_random_particle = [&sensor] {
-        static thread_local auto engine = std::mt19937{std::random_device()()};
-        return beluga::make_from_state<Particle>(sensor.make_random_state(engine));
-      };
-
       particles |= beluga::views::sample |
-                   beluga::views::random_intersperse(make_random_particle, random_state_probability) |
+                   beluga::views::random_intersperse(std::ref(map_distribution), random_state_probability) |
```

Comment:
Just to be sure, we are passing by reference to avoid unnecessarily expensive copies, right?

Reply:
Exactly!